### PR TITLE
add option to prescribe clouds in radiation

### PR DESCRIPTION
**Pull request by szy21**
## Purpose 

Adds an option to prescribe cloud fields in radiation

Also removes nsys for the 4 gpu jobs as it keeps failing.

## Content

----
- [ ] I have read and checked the items on the review checklist.

### CODE REVIEW

**Comment by szy21** [comment]
@Sbozzolo Could you take a look at the code change? Ignore the changes in config or pipeline - I'll revert them.

File: src/solver/model_getters.jl [hunk 1]
```diff
@@ -221,6 +221,9 @@ function get_radiation_mode(parsed_args, ::Type{FT}) where {FT}
     @assert idealized_h2o in (true, false)
     idealized_clouds = parsed_args["idealized_clouds"]
     @assert idealized_clouds in (true, false)
+    prescribed_clouds = parsed_args["prescribed_clouds_in_radiation"]
+    @assert prescribed_clouds in (true, false)
+    @assert !(idealized_clouds && prescribed_clouds)
```
Comment (Sbozzolo) [inline src/solver/model_getters.jl:226]: Can you add an error message when this happens?

Can also add a message to warn when this is activated but the radiation model is not RRTGMP?

File: src/parameterized_tendencies/radiation/RRTMGPInterface.jl [hunk 2]
```diff
@@ -32,6 +33,7 @@ end
 struct AllSkyRadiationWithClearSkyDiagnostics <: AbstractRRTMGPMode
     idealized_h2o::Bool
     idealized_clouds::Bool
+    prescribed_clouds::Bool
```
Comment (Sbozzolo) [inline src/parameterized_tendencies/radiation/RRTMGPInterface.jl:36]: Can you add docstring to describe this?

File: src/parameterized_tendencies/radiation/RRTMGPInterface.jl [hunk 3]
```diff
@@ -24,6 +26,7 @@ end
 struct AllSkyRadiation <: AbstractRRTMGPMode
     idealized_h2o::Bool
     idealized_clouds::Bool
+    cloud::AbstractCloud
     add_isothermal_boundary_layer::Bool
     aerosol_radiation::Bool
```
Comment (Sbozzolo) [inline src/parameterized_tendencies/radiation/RRTMGPInterface.jl:31]:
```suggestion
struct AllSkyRadiation{AC <: AbstractCloud} <: AbstractRRTMGPMode
    idealized_h2o::Bool
    idealized_clouds::Bool
    cloud::AC
    add_isothermal_boundary_layer::Bool
    aerosol_radiation::Bool
```
the struct has to be concrete to be used on a GPU.

I also suggest you pick a name that makes it clear that this is only for radiation.

File: config/default_configs/default_config.yml [hunk 4]
```diff
@@ -99,6 +99,9 @@ dt_rad:
 idealized_clouds:
   help: "Use idealized clouds in radiation model [`false` (default), `true`]"
   value: false
+prescribed_clouds_in_radiation:
+  help: "Use prescribed clouds in radiation model [`false` (default), `true`]"
```
Comment (Sbozzolo) [inline config/default_configs/default_config.yml:103]:
```suggestion
  help: "Use prescribed clouds in radiation model. Clouds are read from ERA5 data and updated every time radiation is called. The year 2021 is used and continuously repeated. This mode only affect radiation and is only relevant for the RRTGMP mode. [`false` (default), `true`]"
```

File: Artifacts.toml [hunk 5]
```diff
@@ -25,3 +25,13 @@ git-tree-sha1 = "6693f8451b447543952b3d557d733c7e5c670ed3"
...
+
+[era5_cloud]
+git-tree-sha1 = "10742e0a2e343d13bb04df379e300a83402d4106"
+
+    [[era5_cloud.download]]
+    sha256 = "bb51e2f2d315b487e05a8d38944d4ad937ee4a40c43b68541220c5d54425e24a"
+    url = "https://caltech.box.com/shared/static/b6ur4ap4vo04j09vdulem96z9fxqlgyn.gz"
+
+
+
```
Comment (Sbozzolo) [inline Artifacts.toml:37]:
```suggestion
```

File: src/solver/types.jl [hunk 6]
```diff
@@ -65,6 +65,26 @@ Refer to ClimaArtifacts for more information on how to obtain the artifact.
 """
 struct PrescribedOzone <: AbstractOzone end
 
+"""
+    AbstractCloud
+
+Describe how cloud properties should be set in radiation.
```
Comment (Sbozzolo) [inline src/solver/types.jl:71]:
```suggestion
Describe how cloud properties should be set in radiation.

This is only relevant for RRTGMP.
```

File: config/model_configs/sphere_aquaplanet_rhoe_equilmoist_allsky_gw_res.yml [hunk 7]
```diff
@@ -3,7 +3,7 @@ z_elem: 31
 dz_bottom: 50.0
 rayleigh_sponge: true
 dt: "400secs"
-t_end: "1days"
```
Comment (szy21) [inline config/model_configs/sphere_aquaplanet_rhoe_equilmoist_allsky_gw_res.yml:6]: revert this

File: .buildkite/longruns_gpu/pipeline.yml
Comment (szy21) [inline .buildkite/longruns_gpu/pipeline.yml:1]: revert this file

File: config/longrun_configs/amip_target_diagedmf.yml [hunk 8]
```diff
@@ -17,6 +17,7 @@ insolation: "timevarying"
 prescribe_ozone: true
 aerosol_radiation: true
 prescribed_aerosols: ["CB1", "CB2", "DST01", "OC1", "OC2", "SO4", "SSLT01"]
+prescribe_clouds_in_radiation: true
```
Comment (szy21) [inline config/longrun_configs/amip_target_diagedmf.yml:20]: revert this

File: config/default_configs/default_config.yml [hunk 9]
```diff
@@ -99,6 +99,9 @@ dt_rad:
 idealized_clouds:
   help: "Use idealized clouds in radiation model [`false` (default), `true`]"
   value: false
+prescribe_clouds_in_radiation:
+  help: "Use prescribed clouds in radiation model. Clouds are read from ERA5 data and updated every time radiation is called. The year 2010 is used and continuously repeated. This mode only affect radiation and is only relevant for the RRTGMP mode. [`false` (default), `true`]"
```
Comment (akshaysridhar) [inline config/default_configs/default_config.yml:103]: @szy21 I think there is a typo here - above in `NEWS.md` it says monthly data from 2020 is used whereas here the 2010 dataset is indicated.

Comment (szy21) [reply config/default_configs/default_config.yml:103]: Thanks. It's a typo in the NEWs. I'll update it.

**Pull request merged**